sdk/monitor/azure-monitor-query/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,208 @@
+# Troubleshooting Azure Monitor Query client library issues
+
+This troubleshooting guide contains instructions to diagnose frequently encountered issues while using the Azure 
+Monitor Query client library for Java. 
+
+## Table of contents
+
+* [General Troubleshooting](#general-troubleshooting)
+    * [Enable client logging](#enable-client-logging)
+    * [Enable HTTP request/response logging](#enable-http-requestresponse-logging)
+    * [Troubleshooting authentication issues with logs and metrics query requests](#authentication-errors)
+* [Troubleshooting Logs Query](#troubleshooting-logs-query)
+    * [Troubleshooting insufficient access error](#troubleshooting-insufficient-access-error)
+    * [Troubleshooting invalid Kusto query](#troubleshooting-invalid-Kusto-query)
+    * [Troubleshooting empty log query results](#troubleshooting-empty-log-query-results)
+    * [Troubleshooting client timeouts when executing logs query request](#troubleshooting-client-timeouts-when-executing-logs-query-request)
+    * [Troubleshooting server timeouts when executing logs query request](#troubleshooting-server-timeouts-when-executing-logs-query-request)
+    * [Troubleshooting server timeouts on OkHTTP client](#troubleshooting-server-timeouts-on-okhttp-client)
+    * [Troubleshooting partially successful logs query requests](#troubleshooting-partially-successful-logs-query-requests)
+* [Troubleshooting Metrics Query](#troubleshooting-metrics-query)
+    * [Troubleshooting unsupported granularity for metrics query](#troubleshooting-unsupported-granularity-for-metrics-query)
+
+## General Troubleshooting
+
+### Enable client logging
```

Comment:
Consider pulling this out into an azure-core/TROUBLESHOOTING.md and link to these SDK-agnostic common steps

Reply:
This is pointing to the msdocs we have for logging which contains a detailed write-up of how to enable logging. Extracting it to azure-core/TROUBLESHOOTING might be needed as the logging documentation is centralized.